src/lint.rs

```diff
@@ -214,6 +246,56 @@ fn lint_impl(tree: &Tree, code: &[u8], lint: &Lint) -> Result<Vec<LintMatch>> {
 /// assert_eq!(matches.len(), 1);
 /// ```
 pub fn lint_custom<'l, I, L>(code: &[u8], lints: I) -> Result<Vec<LintMatch>>
+where
+    I: IntoIterator<Item = L>,
+    L: AsRef<Lint> + 'l,
+{
+    lint_custom_opts(code, lints, &LintOpts::default())
+}
+
+/// Lint code using the provided set of lints with custom options.
+///
+/// Matches are reported in source code order.
+///
+/// - `code` is the source code in question, for example as read from a
+///   file
+/// - `lints` the lints to use for linting the provided source code
+/// - `lint_opts` configuration options for linting behavior
+///
+/// # Examples
+/// ```rust
+/// # use bpflint::builtin_lints;
+/// # use bpflint::lint_custom_opts;
+/// # use bpflint::Lint;
+/// # use bpflint::LintOpts;
+/// let bpf_printk = Lint {
+///     name: "bpf_printk-usage".to_string(),
+///     code: r#"
+///         (call_expression
+///             function: (identifier) @function (#eq? @function "bpf_printk")
+///         )
+///       "#.to_string(),
+///     message: "use bpf_printk only for debugging!".to_string(),
+/// };
+///
+/// let code = br#"
+///     SEC("tp_btf/sched_switch")
+///     int handle__sched_switch(u64 *ctx) {
+///         bpf_printk("context %p\n", ctx);
+///         return 0;
+///     }
+/// "#;
+///
+/// // We want to include the built-in lints as well, not just our
+/// // `bpf_printk` usage flagger.
+/// let matches = lint_custom_opts(code, builtin_lints().chain([bpf_printk]), &LintOpts::default()).unwrap();
+/// assert_eq!(matches.len(), 1);
+/// ```
```

Comment:
This is mostly a copy and paste of the `lint_custom` documentation and the `LintOpts` don't really do anything here. Let's just simplify this:
```suggestion
/// Lint code using the provided set of lints with custom options.
///
/// This function behaves the same as [`lint_custom`], but allows for
/// additional configuration via a [`LintOpts`] object.
```